### PR TITLE
Add more Test cases for Commons and Logic

### LINKED ISSUE #228: Add more Testcases to increase Code Coverage

### PR DESCRIPTION
**Pull request by Jh123x**
Closes #228 

## Changelog:

Fix Some of the bugs which surfaced during the addition of Test Cases

#### Commons
1. Add test for IndexTest
1. Add test for ConfigTest
1. Created test for GUISettingsTest

###### Commands
1. Add test for CommandResult
1. Add test for ExportCommand
1. Add test for ImportCommand
1. Add test for NewMacroCommandTest

###### Parser
1. Add test for ParserUtil
1. Update test for ParameterTest

###### Predicate
1. Update DatePredicateTest
1. Update FoodContainsKeywordsPredicateTest
1. Update TagContainsKeywordsPredicateTest

### CODE REVIEW

File: src/test/java/jimmy/mcgymmy/commons/core/ConfigTest.java [hunk 1]
```diff
@@ -16,12 +24,55 @@ public void toString_defaultObject_stringReturned() {
         assertEquals(defaultConfigAsString, new Config().toString());
     }
 
+    @Test
+    public void loggerLevelCorrect() {
```
Comment (JoeyChenSmart) [inline src/test/java/jimmy/mcgymmy/commons/core/ConfigTest.java]: do we need this?

File: src/test/java/jimmy/mcgymmy/logic/commands/ImportCommandTest.java [hunk 2]
```diff
@@ -38,6 +45,21 @@ public void validImport_success() {
 
         assertCommandSuccess(importCommand, model1,
                 String.format(ImportCommand.MESSAGE_IMPORT_FOOD_SUCCESS, "typicalFoodMcGymmy.json"), model);
+    }
+
+    @Test
+    public void invalidImport_failure() {
```
Comment (JoeyChenSmart) [inline src/test/java/jimmy/mcgymmy/logic/commands/ImportCommandTest.java]: maybe change title to invalidSomethingImport like the below function

File: src/test/java/jimmy/mcgymmy/logic/parser/parameter/OptionalParameterTest.java [hunk 3]
```diff
@@ -24,7 +29,15 @@ void optionalParameter_storesCorrectValue() throws Exception {
...
     void optionalParameter_keepsParentRawValue() throws Exception {
         OptionalParameter<String> parameter = CommandParserTestUtil.makeDummyOptionalParameter("test", "t");
+        assertEquals(parameter.getRawValue(), OPTIONAL_PARAMETER.getRawValue());
         parameter.setValue("poop");
         assertEquals(parameter.getRawValue(), Optional.of("poop"));
+        assertNotEquals(parameter.getRawValue(), OPTIONAL_PARAMETER.getRawValue());
     }
+
+    @Test
+    public void optionalParameter_required() {
```
Comment (JoeyChenSmart) [inline src/test/java/jimmy/mcgymmy/logic/parser/parameter/OptionalParameterTest.java]: should change to notRequired to be clear maybe

File: src/test/java/jimmy/mcgymmy/commons/core/ConfigTest.java (hunk 1 above)
Comment (Jh123x) [reply src/test/java/jimmy/mcgymmy/commons/core/ConfigTest.java]: Hmm not exactly. It's for testing if the setLogLevel function works correctly.
Is it better if i rename it to setLogLevelTest?

Comment (JoeyChenSmart) [reply src/test/java/jimmy/mcgymmy/commons/core/ConfigTest.java]: i meant i don't think these tests add any value. Maybe we should remove this

checking getters/setters are quite redundant to me imo

Comment (Jh123x) [reply src/test/java/jimmy/mcgymmy/commons/core/ConfigTest.java]: Hmm ok
Removed

**Review by JoeyChenSmart — APPROVED** [review]
LGTM

**Pull request merged**